filter_spatial.json

```diff
@@ -0,0 +1,40 @@
+{
+    "id": "filter_spatial",
+    "summary": "Spatial filter using geometries",
+    "description": "Limits the data cube over the spatial dimensions to the specified geometries.\n\n- For **polygons**, the filter retains a pixel in the data cube if the point at the pixel center intersects with at least one of the polygons (as defined in the Simple Features standard by the OGC).\n- For **points** and **lines** (line strings), the filter retains all the pixel centers closest to any of the geometries.\n\nMore specifically, pixels outside of the bounding box of the given geometry will not be available after filtering. All pixels inside the bounding box that are not retained will be set to `null` (no data).",
```

Comment:
My idea was to only have it work as a crop, so the masking part could be optional (we have a separate process for that, indeed). That might bring significant speed improvements because masking is often times expensive.

Reply:
also see discussion at https://github.com/Open-EO/openeo-processes/pull/131#discussion_r368863643

if it is only about cropping: what is the added value of filter_spatial compared to filter_bbox?

Reply:
Cropping only works for a rectangles, but there might be values outside a non-rectangular geometry, which need to be set to null.

Reply:
@soxofaan there is no special added value; my idea is that `filter_spatial` is simply a generalisation of `filter_bbox` that can take more than just a bbox as an input.
@m-mohr Values outside the polygon are just not interesting for the user, but it doesn't meant hat they have to be null. The user just doesn't care about them.

Reply:
Indeed, but we are defining interoperable processes here and thus need to produce comparable results, which means we need to define such things. It's not desirable that some back-ends may set the parts to null and others leave them in.

Reply:
@soxofaan filter_bbox is an easy way to specify a bounding box without needing to mess around with GeoJSON geometries. I just reaalized that the major difference between filter_bbox and filter_spatial is that filter_spatial only works on WGS84 (due to GeoJSON) and filter_bbox allows different CRS. This difference will be weakened once vector cubes are defined though.